internal/resources/metal/gateway/resource.go

```diff
@@ -144,34 +166,37 @@ func (r *Resource) Delete(ctx context.Context, req resource.DeleteRequest, resp
 	id := state.ID.ValueString()
 
 	// API call to delete the Metal Gateway
-	deleteResp, err := client.MetalGateways.Delete(id)
+	_, deleteResp, err := client.MetalGatewaysApi.DeleteMetalGateway(ctx, id).Include(includes).Execute()
```

Comment:
Although it doesn't affect the result, is there any reason to specify Includes in the delete request?

Reply:
Good catch; the includes are needed with the generated SDK because it validates responses based on the spec, and the response will fail validation without these included fields.  I added a comment to document that as well.